EIPS/eip-7637.md

```diff
@@ -0,0 +1,94 @@
+---
+eip: 7637
+title: EXTCODEHASH optimize
+description: Modify the output value of a situation in EXTCODEHASH
+author: Jame (@ZWJKFLC)
+discussions-to: https://ethereum-magicians.org/t/eip-7637-extcodehash-optimize/18946
+status: Draft
+type: Standards Track
+category: Core
+created: 2024-02-26
+---
+
+
+
+## Abstract
+
+This EIP proposal is an optimization for [EIP-1052](./eip-1052.md),
+For addresses with balance but the code is still 0x, the codehash should still be 0x.
+
+
+## Motivation
+
+`EIP-1052` was proposed to save gas fees.
+In order to include the role of `BALANCE`, let the codehash of the address without balance be 0x, and the codehash of the address with balance be hash(0x).
+The contract address can be calculated in advance. Whether it is create or create2, it is possible that the contract is not created but has a balance. For security, you can actually only use keccak256(add.code) == keccak256(bytes(" ")) instead of add.codehash == 0, which makes the original intention of EIP-1052 meaningless.
```

Comment:
```suggestion
The contract address can be calculated in advance. Whether it is `CREATE` or `CREATE2`, it is possible that the contract is not created but has a balance. For security, you can actually only use `keccak256(add.code) == keccak256(bytes(" "))` instead of `add.codehash == 0`, which makes the original intention of EIP-1052 meaningless.
```